### PR TITLE
Add Track Phi Histogram

### PR DESCRIPTION
**Pull request by haandrew**

### CODE REVIEW

**Comment by raymondEhlers** [comment]
+test

**Comment by haandrew** [comment]
@raymondEhlers  please can you commit this now.

File: PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskRecoilJetYield.cxx [hunk 1]
```diff
@@ -962,11 +969,11 @@ Double_t AliAnalysisTaskRecoilJetYield::PTD(AliEmcalJet *Jet, Int_t JetContNb){
       EmcalJetTrackPt[i]=fTrk->Pt();
       EmcalJetNTracks++;
     }
-  fastjet::JetDefinition                *fJetDef;         
+  fastjet::JetDefinition                *fJetDef;
   fastjet::ClusterSequence              *fClustSeqSA;
 
 
-  fJetDef = new fastjet::JetDefinition(fastjet::antikt_algorithm, fJetRadius*2, static_cast<fastjet::RecombinationScheme>(0), fastjet::BestFJ30 ); 
+  fJetDef = new fastjet::JetDefinition(fastjet::antikt_algorithm, fJetRadius*2, static_cast<fastjet::RecombinationScheme>(0), fastjet::BestFJ30 );
```
Comment (mfasDa) [inline PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskRecoilJetYield.cxx]: Memory leak

File: PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskRecoilJetYield.cxx [hunk 2]
```diff
@@ -962,11 +969,11 @@ Double_t AliAnalysisTaskRecoilJetYield::PTD(AliEmcalJet *Jet, Int_t JetContNb){
...
 
   try {
     fClustSeqSA = new fastjet::ClusterSequence(fInputVectors, *fJetDef);
```
Comment (mfasDa) [inline PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskRecoilJetYield.cxx]: Memory leak

File: PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskRecoilJetYield.cxx [hunk 3]
```diff
@@ -1046,7 +1053,7 @@ Double_t AliAnalysisTaskRecoilJetYield::PTD(AliEmcalJet *Jet, Int_t JetContNb){
   fastjet::contrib::Recluster *recluster;
   if(fReclusterAlgo == 2) recluster = new fastjet::contrib::Recluster(fastjet::kt_algorithm,1,true);
   if(fReclusterAlgo == 1) recluster = new fastjet::contrib::Recluster(fastjet::antikt_algorithm,1,true);
-  if(fReclusterAlgo == 0) recluster = new fastjet::contrib::Recluster(fastjet::cambridge_algorithm,1,true);  
+  if(fReclusterAlgo == 0) recluster = new fastjet::contrib::Recluster(fastjet::cambridge_algorithm,1,true);
```
Comment (mfasDa) [inline PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskRecoilJetYield.cxx]: Memory leak

File: PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskRecoilJetYield.cxx [hunk 4]
```diff
@@ -1293,14 +1300,14 @@ Int_t AliAnalysisTaskRecoilJetYield::SelectTriggerHadron(Float_t PtMin, Float_t
...
 	}
       }
-    } 
+    }
   }
   if (Trigger_Counter == 0) return -99999;
-  Int_t RandomNumber = 0, Index = 0 ; 
-  TRandom3* Random = new TRandom3(0); 
+  Int_t RandomNumber = 0, Index = 0 ;
+  TRandom3* Random = new TRandom3(0);
```
Comment (mfasDa) [inline PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskRecoilJetYield.cxx]: Memory leak.

**Comment by haandrew** [comment]
Hi @mfasDa , thanks for pointing out. The leaks should be fixed now

File: PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskRecoilJetYield.cxx [hunk 5]
```diff
@@ -1143,6 +1143,9 @@ Double_t AliAnalysisTaskRecoilJetYield::PTD(AliEmcalJet *Jet, Int_t JetContNb){
   if(!fTruthJet) fJetInfoVar[18]=(finaljet.m());
   else fJetInfoVar[19]=(finaljet.m());
 
+
+  if(fJetDef) delete fJetDef;
+  if(recluster) delete recluster;
```
Comment (mfasDa) [inline PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskRecoilJetYield.cxx]: I think both recluster and fJetDef can be constructed on the stack as the type is known at compile time and the object is not shared with other occurrences. For recluster you make a switch of the fReclusterAlgo and construct the object afterwards. Once the object is on the stack they are destructed automatically once they go out-of-scope.

Comment (haandrew) [reply PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskRecoilJetYield.cxx]: Okay it works to create on the stack for TRandom in the other case that you highlighted but for fJetDef and recluster they are passed to functions that require a type fastjet::contrib:Recluster * for example. Is it possible to do this on the stack?

**Comment by mfasDa** [comment]
+1

**Pull request merged**